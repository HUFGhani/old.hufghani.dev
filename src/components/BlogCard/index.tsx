import { Flex, Heading, Text } from '@chakra-ui/react'
import GatsbyLink from 'gatsby-link'
import React from 'react'

interface Node {
  slug: string
  title: string
  excerpt: string
  publishedAt: Date
}

interface AllGraphCmsPost {
  nodes: Node[]
}

interface BlogCardDataInterface {
  allGraphCmsPost: AllGraphCmsPost
  pagelocation: string
}

const BlogCard: React.FC<BlogCardDataInterface> = ({ allGraphCmsPost, pagelocation }) => {
  return (
    <Flex direction="row" overflow="hidden" overflowX="auto" width="100%" justify="space-evenly">
      {allGraphCmsPost.nodes.map((post: { title: string; excerpt: string; slug: string }, key: any) => {
        return (
          <Flex
            key={key}
            direction="column"
            maxW="md"
            borderWidth="1px"
            borderRadius="lg"
            marginBottom="8"
            width="280px"
            wordBreak="break-word"
            textAlign="center"
            alignContent="center"
            justifyContent="space-between"
            alignItems="center"
            paddingX="2"
            paddingY="2"
          >
            {pagelocation === 'blog' ? (
              <>
                <GatsbyLink to={`${post.slug}`}>
                  <Heading as="h3" fontSize="md" paddingBottom="1">
                    {post.title}
                  </Heading>
                </GatsbyLink>
                <Text noOfLines={3}>{post.excerpt}</Text>
              </>
            ) : (
              <>
                <GatsbyLink to={`blog/${post.slug}`}>
                  <Heading as="h3" fontSize="md" paddingBottom="1">
                    {post.title}
                  </Heading>
                </GatsbyLink>
                <Text noOfLines={3}>{post.excerpt}</Text>
              </>
            )}
          </Flex>
        )
      })}
    </Flex>
  )
}

export default BlogCard
