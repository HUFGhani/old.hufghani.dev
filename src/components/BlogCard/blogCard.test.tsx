import { ChakraProvider, CSSReset } from '@chakra-ui/react'
import { render } from '@testing-library/react'
import React from 'react'
import BlogCard from '.'

const data = {
  nodes: [
    {
      slug: 'test',
      title: 'test',
      excerpt: 'Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...',
      publishedAt: new Date('2022-01-14T19:03:07.828478+00:00'),
    },
  ],
}

const target = 'https://example.com/'

const renderBlogCard = (location: Location) => {
  return render(
    <ChakraProvider>
      <CSSReset />
      <BlogCard allGraphCmsPost={data} pagelocation={location} />
    </ChakraProvider>
  )
}

describe(`<Footer/>`, () => {
  const { location } = window

  beforeAll(() => {
    delete window.location
    // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
    // @ts-ignore
    window.location = {
      pathname: '',
    }
  })
  afterAll((): void => {
    window.location = location
  })

  it(`should render Blog card on index page`, () => {
    window.location.pathname = ''
    const { asFragment } = renderBlogCard(window.location)
    expect(asFragment()).toMatchSnapshot()
  })

  it(`should render Blog card on blog page`, () => {
    window.location.pathname = '/blog'
    const { asFragment } = renderBlogCard(window.location)
    expect(asFragment()).toMatchSnapshot()
  })
})
