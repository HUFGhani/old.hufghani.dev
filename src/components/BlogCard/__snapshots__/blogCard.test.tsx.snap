// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer/> should render Blog card on blog page 1`] = `
<DocumentFragment>
  <div
    class="css-p7ojwx"
  >
    <div
      class="css-18lk41j"
    >
      <a
        href="/test"
      >
        <h3
          class="chakra-heading css-zcv43y"
        >
          test
        </h3>
      </a>
      <p
        class="chakra-text css-1cp6yu0"
      >
        Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Footer/> should render Blog card on index page 1`] = `
<DocumentFragment>
  <div
    class="css-p7ojwx"
  >
    <div
      class="css-18lk41j"
    >
      <a
        href="/blog/test"
      >
        <h3
          class="chakra-heading css-zcv43y"
        >
          test
        </h3>
      </a>
      <p
        class="chakra-text css-1cp6yu0"
      >
        Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...
      </p>
    </div>
  </div>
</DocumentFragment>
`;
